# I. Preparación del fichero población 2020

## I.1. Importamos las librerías necesarias

In [2]:
from functions import shorten_column_titles
import pandas as pd

## I.2. Importamos y exploramos el fichero 
Fuente: https://statbel.fgov.be/nl/open-data/bevolking-naar-woonplaats-nationaliteit-burgerlijke-staat-leeftijd-en-geslacht-12
El fichero existe en dos formatos: xml (dado el volumen, se nos bloquea la descarga a excel online (máx. 25 MB) por la web => queda solo la opción del formato txt)

In [3]:
pobl_2020 = pd.read_csv("TF_SOC_POP_STRUCT_2020.txt", sep = "|")
pobl_2020

CD_REFNIS TX_DESCR_NL TX_DESCR_FR  CD_DSTR_REFNIS  \
0           11001  Aartselaar  Aartselaar           11000   
1           11001  Aartselaar  Aartselaar           11000   
2           11001  Aartselaar  Aartselaar           11000   
3           11001  Aartselaar  Aartselaar           11000   
4           11001  Aartselaar  Aartselaar           11000   
...           ...         ...         ...             ...   
465652      93090   Viroinval   Viroinval           93000   
465653      93090   Viroinval   Viroinval           93000   
465654      93090   Viroinval   Viroinval           93000   
465655      93090   Viroinval   Viroinval           93000   
465656      93090   Viroinval   Viroinval           93000   

                TX_ADM_DSTR_DESCR_NL             TX_ADM_DSTR_DESCR_FR  \
0           Arrondissement Antwerpen          Arrondissement d’Anvers   
1           Arrondissement Antwerpen          Arrondissement d’Anvers   
2           Arrondissement Antwerpen          Arrondissement d’Anvers   
3           Arrondissement Antwerpen          Arrondissement d’Anvers   
4           Arrondissement Antwerpen          Arrondissement d’Anvers   
...                              ...                              ...   
465652  Arrondissement Philippeville  Arrondissement de Philippeville   
465653  Arrondissement Philippeville  Arrondissement de Philippeville   
465654  Arrondissement Philippeville  Arrondissement de Philippeville   
465655  Arrondissement Philippeville  Arrondissement de Philippeville   
465656  Arrondissement Philippeville  Arrondissement de Philippeville   

        CD_PROV_REFNIS     TX_PROV_DESCR_NL   TX_PROV_DESCR_FR  CD_RGN_REFNIS  \
0              10000.0  Provincie Antwerpen  Province d’Anvers           2000   
1              10000.0  Provincie Antwerpen  Province d’Anvers           2000   
2              10000.0  Provincie Antwerpen  Province d’Anvers           2000   
3              10000.0  Provincie Antwerpen  Province d’Anvers           2000   
4              10000.0  Provincie Antwerpen  Province d’Anvers           2000   
...                ...                  ...                ...            ...   
465652         90000.0      Provincie Namen  Province de Namur           3000   
465653         90000.0      Provincie Namen  Province de Namur           3000   
465654         90000.0      Provincie Namen  Province de Namur           3000   
465655         90000.0      Provincie Namen  Province de Namur           3000   
465656         90000.0      Provincie Namen  Province de Namur           3000   

        ...  TX_RGN_DESCR_FR CD_SEX CD_NATLTY TX_NATLTY_NL TX_NATLTY_FR  \
0       ...  Région flamande      M       BEL       Belgen       Belges   
1       ...  Région flamande      M       BEL       Belgen       Belges   
2       ...  Région flamande      F       BEL       Belgen       Belges   
3       ...  Région flamande      M       BEL       Belgen       Belges   
4       ...  Région flamande      M       BEL       Belgen       Belges   
...     ...              ...    ...       ...          ...          ...   
465652  ...  Région wallonne      M       BEL       Belgen       Belges   
465653  ...  Région wallonne      F       BEL       Belgen       Belges   
465654  ...  Région wallonne      F       BEL       Belgen       Belges   
465655  ...  Région wallonne      M       BEL       Belgen       Belges   
465656  ...  Région wallonne      F       BEL       Belgen       Belges   

       CD_CIV_STS  TX_CIV_STS_NL TX_CIV_STS_FR CD_AGE  MS_POPULATION  
0               4     Gescheiden       Divorcé     60             17  
1               4     Gescheiden       Divorcé     84              2  
2               4     Gescheiden       Divorcé     55             17  
3               4     Gescheiden       Divorcé     59             15  
4               4     Gescheiden       Divorcé     35              2  
...           ...            ...           ...    ...            ...  
465652          2         G

In [4]:
pobl_2020.shape

(465657, 21)

In [5]:
pobl_2020.describe(include="all")

CD_REFNIS TX_DESCR_NL TX_DESCR_FR  CD_DSTR_REFNIS  \
count   465657.000000      465657      465657   465657.000000   
unique            NaN         581         581             NaN   
top               NaN   Antwerpen      Anvers             NaN   
freq              NaN        1318        1318             NaN   
mean     45734.611341         NaN         NaN    45693.373019   
std      24010.222862         NaN         NaN    24004.408961   
min      11001.000000         NaN         NaN    11000.000000   
25%      24028.000000         NaN         NaN    24000.000000   
50%      44045.000000         NaN         NaN    44000.000000   
75%      63035.000000         NaN         NaN    63000.000000   
max      93090.000000         NaN         NaN    93000.000000   

                  TX_ADM_DSTR_DESCR_NL            TX_ADM_DSTR_DESCR_FR  \
count                           465657                          465657   
unique                              43                              43   
top     Arrondissement Halle-Vilvoorde  Arrondissement de Hal-Vilvorde   
freq                             29500                           29500   
mean                               NaN                             NaN   
std                                NaN                             NaN   
min                                NaN                             NaN   
25%                                NaN                             NaN   
50%                                NaN                             NaN   
75%                                NaN                             NaN   
max                                NaN                             NaN   

        CD_PROV_REFNIS TX_PROV_DESCR_NL   TX_PROV_DESCR_FR  CD_RGN_REFNIS  \
count    444165.000000           444165             444165  465657.000000   
unique             NaN               10                 10            NaN   
top                NaN   Provincie Luik  Province de Liège            NaN   
freq               NaN            62491              62491            NaN   
mean      43753.283940              NaN                NaN    2521.016542   
std       24151.307584              NaN                NaN     584.694216   
min       10000.000000              NaN                NaN    2000.000000   
25%       20001.000000              NaN                NaN    2000.000000   
50%       40000.000000              NaN                NaN    2000.000000   
75%       60000.000000              NaN                NaN    3000.000000   
max       90000.000000              NaN                NaN    4000.000000   

        ...  TX_RGN_DESCR_FR  CD_SEX CD_NATLTY TX_NATLTY_NL TX_NATLTY_FR  \
count   ...           465657  465657    465657       465657       465657   
unique  ...                3       2         2            2            2   
top     ...  Région flamande       F       BEL       Belgen       Belges   
freq    ...           244534  239897    306935       306935       306935   
mean    ...              NaN     NaN       NaN          NaN          NaN   
std     ...              NaN     NaN       NaN          NaN          NaN   
min     ...              NaN     NaN       NaN          NaN          NaN   
25%     ...              NaN     NaN       NaN          NaN          NaN   
50%     ...              NaN     NaN       NaN          NaN          NaN   
75%     ...              NaN     NaN       NaN          NaN          NaN   
max     ...              NaN     NaN       NaN          NaN          NaN   

           CD_CIV_STS  TX_CIV_STS_NL TX_CIV_STS_FR         CD_AGE  \
count   465657.000000         465657        465657  465657.000000   
unique            NaN              4             4            NaN   
top               NaN       Ongehuwd   Célibataire            NaN   
freq              NaN         162780        162780            NaN   
mean         2.207402            NaN           NaN      52.942507   
std          1.124354            NaN           NaN      23.585657   
min          1.000000

In [6]:
pobl_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 465657 entries, 0 to 465656
Data columns (total 21 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   CD_REFNIS             465657 non-null  int64  
 1   TX_DESCR_NL           465657 non-null  object 
 2   TX_DESCR_FR           465657 non-null  object 
 3   CD_DSTR_REFNIS        465657 non-null  int64  
 4   TX_ADM_DSTR_DESCR_NL  465657 non-null  object 
 5   TX_ADM_DSTR_DESCR_FR  465657 non-null  object 
 6   CD_PROV_REFNIS        444165 non-null  float64
 7   TX_PROV_DESCR_NL      444165 non-null  object 
 8   TX_PROV_DESCR_FR      444165 non-null  object 
 9   CD_RGN_REFNIS         465657 non-null  int64  
 10  TX_RGN_DESCR_NL       465657 non-null  object 
 11  TX_RGN_DESCR_FR       465657 non-null  object 
 12  CD_SEX                465657 non-null  object 
 13  CD_NATLTY             465657 non-null  object 
 14  TX_NATLTY_NL          465657 non-null  object 
 15  

## I.3. Limpiamos el dataframe

### 1.3.1. Quitamos las columnas irrelevantes para nuestro propósito
Los únicos datos requeridos son: número de habitantes por "arrondissement" => las únicas columnas por guardar son: TX_PROV_DESCR_NL, TX_ADM_DSTR_DESCR_NL + MS_POPULATION

In [7]:
pobl_2020_col = pobl_2020.loc[:,["TX_PROV_DESCR_NL", "TX_ADM_DSTR_DESCR_NL","MS_POPULATION"]]
pobl_2020_col.head()

TX_PROV_DESCR_NL      TX_ADM_DSTR_DESCR_NL  MS_POPULATION
0  Provincie Antwerpen  Arrondissement Antwerpen             17
1  Provincie Antwerpen  Arrondissement Antwerpen              2
2  Provincie Antwerpen  Arrondissement Antwerpen             17
3  Provincie Antwerpen  Arrondissement Antwerpen             15
4  Provincie Antwerpen  Arrondissement Antwerpen              2

In [8]:
pobl_2020_col.columns

Index(['TX_PROV_DESCR_NL', 'TX_ADM_DSTR_DESCR_NL', 'MS_POPULATION'], dtype='object')

In [9]:
#adaptar los nombres de títulos e integrarlos en el dataframe:
pobl_2020_col.columns = shorten_column_titles(pobl_2020_col.columns)
pobl_2020_col.head()

PROV                  ADM_DSTR  MS_POPULATION
0  Provincie Antwerpen  Arrondissement Antwerpen             17
1  Provincie Antwerpen  Arrondissement Antwerpen              2
2  Provincie Antwerpen  Arrondissement Antwerpen             17
3  Provincie Antwerpen  Arrondissement Antwerpen             15
4  Provincie Antwerpen  Arrondissement Antwerpen              2

### I.3.2. Tratamos los valores no disponibles (emergidos en el I.2)

a) Identificamos las columnas concretas dónde se sitúan los valores no disponibles de una manera más visible que por el método .info() en el I.2.

In [10]:
#guardamos los últimos cambios en la variables pobl_2020
pobl_2020 = pobl_2020_col
pobl_2020_col.isnull().any()

PROV              True
ADM_DSTR         False
MS_POPULATION    False
dtype: bool

b) Apreciamos la cantidad de valores no disponibles (nº de registros en los cuales se manifiestan)

In [11]:
pobl_2020.isnull().sum()

PROV             21492
ADM_DSTR             0
MS_POPULATION        0
dtype: int64

c) Con vistas a la sustitución de estos valores: verificamos primero si el número de 21492 corresponde al número de filas que conciernan solo al arrondissement Brussel-Hoofdstad

In [12]:
#primer sondeo mediante sample:
nan_rows = pobl_2020[pobl_2020.isnull().any(1)]
nan_rows.sample(5)

PROV                          ADM_DSTR  MS_POPULATION
74043  NaN  Arrondissement Brussel-Hoofdstad              3
62714  NaN  Arrondissement Brussel-Hoofdstad             86
62024  NaN  Arrondissement Brussel-Hoofdstad             15
72719  NaN  Arrondissement Brussel-Hoofdstad             62
67717  NaN  Arrondissement Brussel-Hoofdstad             48

In [13]:
#cuento exacto:
sum(pobl_2020['ADM_DSTR'] == 'Arrondissement Brussel-Hoofdstad')

21492

Conclusión: el número de filas sin valor disponible corresponde exactamente al número de filas que contienen "Arrondissement Brussel-Hoofdstad", ni más, ni menos. Por lo tanto, sabemos con certeza que todas las sustituciones se situarán en este "arrondissement". Para no perder datos en posteriores visualizaciones o tratamientos de datos en base a 'provincia', vamos a 'copiar' los valores de arrondissement a la columna de provincia. Así, siempre que llamemos a este dataframe (o de manera más concreta, a sus provincias), estaremos seguros de que no hay ningún valor NaN, y entrarán en consideración siempre los datos de la capital, aunque la 'etiqueta' no pueda considerarse propiamente como una provincia.

d) Sustituimos todos los valores "NaN" por "Arr. Brussel-Hoofdstad[no Prov]"

In [14]:
pobl_2020['PROV'] = pobl_2020['PROV'].fillna("Arr. Brussel-Hfdstad[no Prov]")

In [15]:
#verificamos si ahora nuestro dataframe se ha quedado sin valores NaN
pobl_2020.isnull().sum()

PROV             0
ADM_DSTR         0
MS_POPULATION    0
dtype: int64

Ya está limpio pobl_2020: no quedan más valores Nan.

### I.3.3. Agrupamos las filas por 'provincia' y 'arrondissement' para obtener así el nº total de habitantes por 'arrondissement'. Optamos por mantener las provincias para que todos puedan situar con más facilidad los 'arrondissmenten'.

### a) agrupación de número de habitantes a nivel de 'arrondissement' 

In [16]:
pobl_2020_habxarr = pobl_2020.groupby(["PROV","ADM_DSTR"])[["PROV","ADM_DSTR", "MS_POPULATION"]].sum()
pobl_2020_habxarr

MS_POPULATION
PROV                          ADM_DSTR                                       
Arr. Brussel-Hfdstad[no Prov] Arrondissement Brussel-Hoofdstad        1218255
Provincie Antwerpen           Arrondissement Antwerpen                1057736
                              Arrondissement Mechelen                  347125
                              Arrondissement Turnhout                  464869
Provincie Henegouwen          Arrondissement Aat                       128468
                              Arrondissement Bergen                    259237
                              Arrondissement Charleroi                 396962
                              Arrondissement Doornik-Moeskroen         223799
                              Arrondissement La Louvière               141470
                              Arrondissement Thuin                      91725
                              Arrondissement Zinnik                    105179
Provincie Limburg             Arrondissement Hasselt                   420312
                              Arrondissement Maaseik                   252115
                              Arrondissement Tongeren                  204943
Provincie Luik                Arrondissement Borgworm                   81889
                              Arrondissement Hoei                      113869
                              Arrondissement Luik                      625765
                              Arrondissement Verviers                  288277
Provincie Luxemburg           Arrondissement Aarlen                     62996
                              Arrondissement Bastenaken                 49083
                              Arrondissement Marche-en-Famenne          56771
                              Arrondissement Neufchâteau                63763
                              Arrondissement Virton                     54139
Provincie Namen               Arrondissement Dinant                    111286
                              Arrondissement Namen                     318231
                              Arrondissement Philippeville              66315
Provincie Oost-Vlaanderen     Arrondissement Aalst                     293650
                              Arrondissement Dendermonde               202411
                              Arrondissement Eeklo                      85692
                              Arrondissement Gent                      564042
                              Arrondissement Oudenaarde                124610
                              Arrondissement Sint-Niklaas              254850
Provincie Vlaams-Brabant      Arrondissement Halle-Vilvoorde           643766
                              Arrondissement Leuven                    512077
Provincie Waals-Brabant       Arrondissement Nijvel                    406019
Provincie West-Vlaanderen     Arrondissement Brugge                    282745
                              Arrondissement Diksmuide                  51696
                              Arrondissement Ieper                     106570
                              Arrondissement Kortrijk                  292493
                              Arrondissement Oostende                  157780
                              Arrondissement Roeselare                 154494
                              Arrondissement Tielt                      93428
                              Arrondissement Veurne                     61739

In [17]:
#reset index para merge en base a columnas idénticas
pobl_2020_habxarr.reset_index(inplace=True)

In [18]:
#verificación: nuevo índice? sí
pobl_2020_habxarr

PROV                          ADM_DSTR  \
0   Arr. Brussel-Hfdstad[no Prov]  Arrondissement Brussel-Hoofdstad   
1             Provincie Antwerpen          Arrondissement Antwerpen   
2             Provincie Antwerpen           Arrondissement Mechelen   
3             Provincie Antwerpen           Arrondissement Turnhout   
4            Provincie Henegouwen                Arrondissement Aat   
5            Provincie Henegouwen             Arrondissement Bergen   
6            Provincie Henegouwen          Arrondissement Charleroi   
7            Provincie Henegouwen  Arrondissement Doornik-Moeskroen   
8            Provincie Henegouwen        Arrondissement La Louvière   
9            Provincie Henegouwen              Arrondissement Thuin   
10           Provincie Henegouwen             Arrondissement Zinnik   
11              Provincie Limburg            Arrondissement Hasselt   
12              Provincie Limburg            Arrondissement Maaseik   
13              Provincie Limburg           Arrondissement Tongeren   
14                 Provincie Luik           Arrondissement Borgworm   
15                 Provincie Luik               Arrondissement Hoei   
16                 Provincie Luik               Arrondissement Luik   
17                 Provincie Luik           Arrondissement Verviers   
18            Provincie Luxemburg             Arrondissement Aarlen   
19            Provincie Luxemburg         Arrondissement Bastenaken   
20            Provincie Luxemburg  Arrondissement Marche-en-Famenne   
21            Provincie Luxemburg        Arrondissement Neufchâteau   
22            Provincie Luxemburg             Arrondissement Virton   
23                Provincie Namen             Arrondissement Dinant   
24                Provincie Namen              Arrondissement Namen   
25                Provincie Namen      Arrondissement Philippeville   
26      Provincie Oost-Vlaanderen              Arrondissement Aalst   
27      Provincie Oost-Vlaanderen        Arrondissement Dendermonde   
28      Provincie Oost-Vlaanderen              Arrondissement Eeklo   
29      Provincie Oost-Vlaanderen               Arrondissement Gent   
30      Provincie Oost-Vlaanderen         Arrondissement Oudenaarde   
31      Provincie Oost-Vlaanderen       Arrondissement Sint-Niklaas   
32       Provincie Vlaams-Brabant    Arrondissement Halle-Vilvoorde   
33       Provincie Vlaams-Brabant             Arrondissement Leuven   
34        Provincie Waals-Brabant             Arrondissement Nijvel   
35      Provincie West-Vlaanderen             Arrondissement Brugge   
36      Provincie West-Vlaanderen          Arrondissement Diksmuide   
37      Provincie West-Vlaanderen              Arrondissement Ieper   
38      Provincie West-Vlaanderen           Arrondissement Kortrijk   
39      Provincie West-Vlaanderen           Arrondissement Oostende   
40      Provincie West-Vlaanderen          Arrondissement Roeselare   
41      Provincie West-Vlaanderen              Arrondissement Tielt   
42      Provincie West-Vlaanderen             Arrondissement Veurne   

    MS_POPULATION  
0         1218255  
1         1057736  
2          347125  
3          464869  
4          128468  
5          259237  
6          396962  
7          223799  
8          141470  
9           91725  
10         105179  
11         420312  
12         252115  
13         204943  
14          81889  
15         113869  
16         625765  
17         288277  
18          62996  
19          49083  
20          56771  
21          63763  
22          54139  
23         111286  
24         318231  
25          66315  
26         293650  
27         202411  
28          85692  
29         564042  
30         124610  
31         254850  
32         643766  
33         512077  
34         406019  
35         282745  
36          51696  
37         106570  
38         292493  
39         157780  
40         154494  
41          93428  
42          61739

### b) agrupación de nº de habitantes por provincia (Bxl: arrondissment)

In [19]:
pobl_2020_habxprov = pobl_2020_habxarr.groupby("PROV")[["PROV","MS_POPULATION"]].sum()
pobl_2020_habxprov

MS_POPULATION
PROV                                        
Arr. Brussel-Hfdstad[no Prov]        1218255
Provincie Antwerpen                  1869730
Provincie Henegouwen                 1346840
Provincie Limburg                     877370
Provincie Luik                       1109800
Provincie Luxemburg                   286752
Provincie Namen                       495832
Provincie Oost-Vlaanderen            1525255
Provincie Vlaams-Brabant             1155843
Provincie Waals-Brabant               406019
Provincie West-Vlaanderen            1200945

### c) agrupación de nº de habitantes a nivel de país

In [20]:
hab_2020_BE = pobl_2020["MS_POPULATION"].sum()
hab_2020_BE

11492641

### d) nº de habitantes por "arrondissement": cifras absolutas y proporcionales respecto a la totalidad de la población belga

In [21]:
pobl_2020_habxarr["PROPORTION_BE"] = pobl_2020_habxarr["MS_POPULATION"]/pobl_2020_habxarr["MS_POPULATION"].sum()
#por haber querido cambiar título nueva columna de PROPORCIÓN a PROPORCIÓN_BE, quitar la penúltima columna PROPORCION
#luego lo mismo con PROPORCIÓN_BE > PROPORTION_BE
#pobl_2020_habxarr.drop("PROPORCIÓN", axis=1, inplace=True)
#pobl_2020_habxarr.drop("PROPORCIÓN_BE", axis = 1, inplace=True)
pobl_2020_habxarr

PROV                          ADM_DSTR  \
0   Arr. Brussel-Hfdstad[no Prov]  Arrondissement Brussel-Hoofdstad   
1             Provincie Antwerpen          Arrondissement Antwerpen   
2             Provincie Antwerpen           Arrondissement Mechelen   
3             Provincie Antwerpen           Arrondissement Turnhout   
4            Provincie Henegouwen                Arrondissement Aat   
5            Provincie Henegouwen             Arrondissement Bergen   
6            Provincie Henegouwen          Arrondissement Charleroi   
7            Provincie Henegouwen  Arrondissement Doornik-Moeskroen   
8            Provincie Henegouwen        Arrondissement La Louvière   
9            Provincie Henegouwen              Arrondissement Thuin   
10           Provincie Henegouwen             Arrondissement Zinnik   
11              Provincie Limburg            Arrondissement Hasselt   
12              Provincie Limburg            Arrondissement Maaseik   
13              Provincie Limburg           Arrondissement Tongeren   
14                 Provincie Luik           Arrondissement Borgworm   
15                 Provincie Luik               Arrondissement Hoei   
16                 Provincie Luik               Arrondissement Luik   
17                 Provincie Luik           Arrondissement Verviers   
18            Provincie Luxemburg             Arrondissement Aarlen   
19            Provincie Luxemburg         Arrondissement Bastenaken   
20            Provincie Luxemburg  Arrondissement Marche-en-Famenne   
21            Provincie Luxemburg        Arrondissement Neufchâteau   
22            Provincie Luxemburg             Arrondissement Virton   
23                Provincie Namen             Arrondissement Dinant   
24                Provincie Namen              Arrondissement Namen   
25                Provincie Namen      Arrondissement Philippeville   
26      Provincie Oost-Vlaanderen              Arrondissement Aalst   
27      Provincie Oost-Vlaanderen        Arrondissement Dendermonde   
28      Provincie Oost-Vlaanderen              Arrondissement Eeklo   
29      Provincie Oost-Vlaanderen               Arrondissement Gent   
30      Provincie Oost-Vlaanderen         Arrondissement Oudenaarde   
31      Provincie Oost-Vlaanderen       Arrondissement Sint-Niklaas   
32       Provincie Vlaams-Brabant    Arrondissement Halle-Vilvoorde   
33       Provincie Vlaams-Brabant             Arrondissement Leuven   
34        Provincie Waals-Brabant             Arrondissement Nijvel   
35      Provincie West-Vlaanderen             Arrondissement Brugge   
36      Provincie West-Vlaanderen          Arrondissement Diksmuide   
37      Provincie West-Vlaanderen              Arrondissement Ieper   
38      Provincie West-Vlaanderen           Arrondissement Kortrijk   
39      Provincie West-Vlaanderen           Arrondissement Oostende   
40      Provincie West-Vlaanderen          Arrondissement Roeselare   
41      Provincie West-Vlaanderen              Arrondissement Tielt   
42      Provincie West-Vlaanderen             Arrondissement Veurne   

    MS_POPULATION  PROPORTION_BE  
0         1218255       0.106003  
1         1057736       0.092036  
2          347125       0.030204  
3          464869       0.040449  
4          128468       0.011178  
5          259237       0.022557  
6          396962       0.034541  
7          223799       0.019473  
8          141470       0.012310  
9           91725       0.007981  
10         105179       0.009152  
11         420312       0.036572  
12         252115       0.021937  
13         204943       0.017833  
14          81889       0.007125  
15         113869       0.009908  
16         625765       0.054449  
17         288277       0.025084  
18          62996       0.005481  
19          49083       0.004271  
20          56771       0.004940  
21          63763       0.005548  
22          54139       0.004711  
23         111286       0.009683  
24         318231       0.027

### e) nº de habitantes por "provincia": cifras absolutas y proporcionales respecto a la totalidad de la población belga

In [22]:
pobl_2020_habxprov["PROPORCIÓN_BE"] = pobl_2020_habxprov["MS_POPULATION"]/pobl_2020_habxprov["MS_POPULATION"].sum()
pobl_2020_habxprov

MS_POPULATION  PROPORCIÓN_BE
PROV                                                       
Arr. Brussel-Hfdstad[no Prov]        1218255       0.106003
Provincie Antwerpen                  1869730       0.162689
Provincie Henegouwen                 1346840       0.117192
Provincie Limburg                     877370       0.076342
Provincie Luik                       1109800       0.096566
Provincie Luxemburg                   286752       0.024951
Provincie Namen                       495832       0.043143
Provincie Oost-Vlaanderen            1525255       0.132716
Provincie Vlaams-Brabant             1155843       0.100572
Provincie Waals-Brabant               406019       0.035329
Provincie West-Vlaanderen            1200945       0.104497

## I.4 Preparación de ficheros para exportar



In [34]:
pobl_2020_habxarr.to_csv("Data_ACC/pobl_2020_habxarr.csv", index = False)
pobl_2020_habxprov.to_csv("Data_ACC/pobl_2020_habxprov.csv", index = False)
#cambios para hab_2020_BE porque como numpy.array no admite escritura a csv
hab_2020_BE.tofile('Data_acc/hab_2020_BE.csv', sep = ',')